balancer-js/src/modules/contracts/implementations/ERC20.ts

```diff
@@ -1,9 +1,10 @@
 import { Contract } from '@ethersproject/contracts';
 import { Provider } from '@ethersproject/providers';
 import { Signer } from '@ethersproject/abstract-signer';
-import abi from '../../../lib/abi/ERC20.json';
+import { ERC20__factory } from '@/contracts';
 
 export const ERC20 = (
   address: string,
   signerOrProvider: Signer | Provider
-): Contract => new Contract(address, abi, signerOrProvider);
+): Contract =>
+  new Contract(address, ERC20__factory.createInterface(), signerOrProvider);
```

Comment:
Since we can now import the contract that typechain generates, do we really need these implementation files? I think this refactor is a nice opportunity to evaluate removing them.
We also get the benefit from having the contract type itself (e.g. `ERC20` on this case, instead of `Contract`).